samples/seat_massager/provider/src/provider_impl.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+use digital_twin_model::{sdv_v1 as sdv, Metadata};
+use log::{debug, info, warn};
+use parking_lot::{Mutex, MutexGuard};
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_provider::{
+    digital_twin_provider_server::DigitalTwinProvider, GetRequest, GetResponse, InvokeRequest,
+    InvokeResponse, SetRequest, SetResponse, SubscribeRequest, SubscribeResponse,
+    UnsubscribeRequest, UnsubscribeResponse,
+};
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_consumer::digital_twin_consumer_client::DigitalTwinConsumerClient;
+use samples_protobuf_data_access::sample_grpc::v1::digital_twin_consumer::PublishRequest;
+use serde_derive::{Deserialize, Serialize};
+use std::sync::Arc;
+use std::vec::Vec;
+use tonic::{Request, Response, Status};
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Property {
+    #[serde(rename = "MassageAirbags")]
+    massage_airbags: sdv::airbag_seat_massager::massage_airbags::TYPE,
+    #[serde(rename = "$metadata")]
+    metadata: Metadata,
+}
+
+#[derive(Debug, Default)]
+pub struct ProviderProperties {
+    pub massage_airbags: sdv::airbag_seat_massager::massage_airbags::TYPE,
+}
+
+#[derive(Debug, Default)]
+pub struct ProviderImpl {
+    pub properties: Arc<Mutex<ProviderProperties>>,
+}
+
+impl ProviderImpl {
+    fn set_message_airbags(
+        properties: Arc<Mutex<ProviderProperties>>,
+        value: &str,
+    ) -> Result<(), String> {
+        let message_airbags_property_json: serde_json::Value = serde_json::from_str(value).unwrap();
+        let message_airbags_json = message_airbags_property_json
+            .get(sdv::airbag_seat_massager::massage_airbags::NAME)
+            .unwrap();
+
+        let massage_airbags: sdv::airbag_seat_massager::massage_airbags::TYPE =
+            serde_json::from_value(message_airbags_json.clone()).unwrap();
+
+        info!("Setting message airbags to: {:?}", massage_airbags);
+
+        // This block controls the lifetime of the lock.
+        {
+            let mut lock: MutexGuard<ProviderProperties> = properties.lock();
+
+            lock.massage_airbags = massage_airbags;
+        }
+
+        Ok(())
+    }
+
+    fn get_message_airbags(properties: Arc<Mutex<ProviderProperties>>) -> Result<String, String> {
+        let mut property: Property = Property {
```

Comment:
the type annotation here is not necessary

Reply:
I prefer to make it explicit.  Leaving it as is.